src/main/io/osd.c

```diff
@@ -956,6 +956,14 @@ static bool osdDrawSingleElement(uint8_t item)
                 break;
             }
 
+#ifdef USE_GPS_RESCUE
+            if (gpsRescueIsConfigured() && osdWarnGetState(OSD_WARNING_GPS_RESCUE_UNAVAILABLE) && !isGPSRescueAvailable()) {
+                osdFormatMessage(buff, OSD_FORMAT_MESSAGE_BUFFER_SIZE, "NO GPS RESC");
+                SET_BLINK(OSD_WARNINGS);
+                break;
+            }
+#endif
+
```

Comment:
I think this should be limited to the ARMED state only. Otherwise the warning could come and go while the pilot is  sitting unarmed and not be meaningful.

Reply:
Yes, that will be annoying - will add armed check.  I was also wondering about the gps_rescue_min_dth configuration, i.e. whether it should trigger this warning as well.  Scenario:  flying at the edge of a cliff, you may not realize that GPS rescue is not available because you are within gps_rescue_min_dth.  In the other hand, people may not take off because they think there is something wrong with GPS rescue if the warning comes on after arming because they are within gps_rescue_min_dth.

Reply:
I would stick with just the elements that could cause an intermittent drop-out such as lost sat fix, etc.  Otherwise people will get used to always seeing the warning until they're far away and it will lose it's impact.

I think there's a case for a separate (not the warnings field) element to relay GPS Rescue specific information. Something like this could be used for all GPS Rescue info like "TOO CLOSE", etc.  This field could also show status info like "IDLE", ATTAIN ALT", "CROSSTRACK", etc.  The problem with mixing these into the warnings field is one of priority. Like should a "LOW BATTERY" warning supersede or not, etc.  It gets very difficult to manage so many types warnings in parallel through a single field.